### ISSUE TITLE
Talk 11: Nilesh Mahajan

### ISSUE THREAD

**Issue by rrnewton** [wonks/ICFP_rehearsal_feedback#11]

**Comment by cgswords** [comment]
Put the communication on the slides!

**Comment by ccshan** [comment]
Watch where you're pointing with the laser pointer and be intentional!

**Comment by cgswords** [comment]
Kanor's syntax is hard to read for me. Is it possible to abstract over it maybe, or give pictoral explainations of these (clearly powerful) language constructs?

**Comment by samth** [comment]
Pictures!

**Comment by ccshan** [comment]
At the "Kanor Equivalent" slide I wonder whether the processing of sender and receiver pairs takes O(n^2) rather than O(n) time using Kanor, where n is the number of processors.

**Comment by ccshan** [comment]
Please enlarge the code snippets (perhaps inserting more line breaks)

**Comment by ccshan** [comment]
Why "Allgatherv" in the slide title?

**Comment by cgswords** [comment]
Also, please define the keywords? What does `_at_` do? Maybe it does a lot of things?

**Comment by samth** [comment]
Showing the "formal syntax" is rarely useful -- show examples instead.

**Comment by ccshan** [comment]
I suspect that any point you try to make using a Formal Syntax slide is better made by example without showing any BNF production.

**Comment by samth** [comment]
@ccshan Jinx.

**Comment by cgswords** [comment]
Yes, that. Spend time on smaller examples before diving in, leading us through the language features we need to know before building the big ones.

**Comment by cgswords** [comment]
@samth I nearly typed "and the thing Sam is about to post about not showing the formal definition" ;)

**Comment by cgswords** [comment]
What guarantees do we have that communcation will always succeed?

**Comment by cgswords** [comment]
"Better than what"? You describe these properties and then say we can do better at runtime, but it's hard to tell what you're trying to improve there.

**Comment by cgswords** [comment]
Does the programmer define the communication knowledge of their application? Or is this a thing that crops up based on how you write the program, to be used as a general classification technique?

**Comment by cgswords** [comment]
For your algorithm, it may be better to gave two computation graphs, one where it holds and one where it doesn't, and talk through the algorithm with them, demonstrating how one fails and one succeeds.

**Comment by cgswords** [comment]
What sorts of compiler support are you shooting for? Why is it a problem?

**Comment by cgswords** [comment]
Color key for the tests, and maybe use fewer graphs.

**Comment by samth** [comment]
Have a thank you slide that includes your main points so that it's up during the question period.

**Comment by samth** [comment]
_Feedback_ 
- Dan: avoid reading your slides
- Sam: limit yourself to 5 bullets for the whole talk -- use pictures instead.
- Ken: for each section, write down for yourself what your communication goal is for that section

**Comment by ccshan** [comment]
- Dan: Not having text on your slides helps you not read them.
- Cam: Not reading your slides helps you appear competent.
